src/libImaging/PcxDecode.c

```diff
@@ -60,15 +60,25 @@ ImagingPcxDecode(Imaging im, ImagingCodecState state, UINT8 *buf, Py_ssize_t byt
         }
 
         if (state->x >= state->bytes) {
-            if (state->bytes % state->xsize && state->bytes > state->xsize) {
-                int bands = state->bytes / state->xsize;
-                int stride = state->bytes / bands;
+            int bands;
+            int xsize = 0;
+            int stride = 0;
+            if (state->bits == 2 || state->bits == 4) {
+                xsize = (state->xsize + 7) / 8;
+                bands = state->bits;
+                stride = state->bytes / state->bits;
+            } else {
+                xsize = state->xsize;
+                bands = state->bytes / state->xsize;
```

Comment:
We had a similar caluclation before, but can `state->xsize` ever be zero and cause a divide by zero here?

Reply:
We can't open a zero width image, no. See my explanation at the start of #8408, or a demonstration at https://github.com/radarhere/Pillow/actions/runs/15560051255/job/43809855672#step:11:119